### PR TITLE
Update script reference for advanced script changes

### PR DESCRIPTION
**Pull request by Licenser**

### CODE REVIEW

**Review by mfelsche — COMMENTED** [review]
I do think we should also add a section to https://github.com/tremor-rs/tremor-www/blob/main/docs/language/scripts.md

**Comment by Licenser** [comment]
You are absolutely right, I missed that, we got too much duplication -.-

File: docs/reference/operators/script.md [hunk 1]
```diff
@@ -24,8 +31,11 @@ end;
 
 ```tremor
 define script add
+state
```
[the remark this answers is not in the document]
Comment (Licenser) [reply docs/reference/operators/script.md:34]: put in a full example of a configurable adder

File: docs/language/scripts.md [hunk 2]
```diff
@@ -27,10 +36,21 @@ end;
 
 ## Execution
 
-Scripts are executed and evaluated for each event [Select]ed into its `in` port. They are interpreted statement by statement, at each execution step the [event], the script state and the local variable stack can be manipulated.
+Scripts are executed and evaluated for each event [Select]ed into its `in` port. They are interpreted statement by statement, at each execution step the [event], the script `state` and the local variable stack can be manipulated.
```
Comment (mfelsche) [inline docs/language/scripts.md:39]: :heart_eyes_cat:

File: docs/language/scripts.md [hunk 3]
```diff
@@ -27,10 +36,21 @@ end;
...
 
 During execution, the payload of the current event is available via the special `event` [path](./expressions.md#paths). As on each other level inside the Tremor runtime [events] can be arbitrarily structured values, closely resembling JSON, but with some extensions. See the section on Tremors [Type System](./index.md#type-system).
 
+
+### state
+
+The script state, accessed through the `state` keyword, is a shared, mutable, state that persists across events. It can be initialized with the `state` section in the script definition.
+
+Multiple script sections access the same state allowing to seperate configuration logic from data processing logic.
```
Comment (mfelsche) [inline docs/language/scripts.md:48]:
```suggestion
Multiple script sections access the same state allowing to separate configuration logic from data processing logic.
```

File: docs/language/scripts.md [hunk 4]
```diff
@@ -4,18 +4,27 @@ sidebar_position: 1
 
 # Scripts
 
-Tremor scripts are user-defined pipeline [Operators]. They also have the standard ports `in`, `out` and `err` and are part of the [Pipeline] graph. Events are routed into and out of scripts using the [Select] statement inside a [Pipeline].
+Tremor scripts are user-defined pipeline [Operators]. They also have the standard ports `in`, `out` and `err` and are part of the [Pipeline] graph in addition `select from <port>` allow specifying scripts for additional ports. Events are routed into and out of scripts using the [Select] statement inside a [Pipeline].
```
Comment (mfelsche) [inline docs/language/scripts.md:7]:
```suggestion
Tremor scripts are user-defined pipeline [Operators]. They also have the standard ports `in`, `out` and `err` and are part of the [Pipeline] graph in addition `script from <port>` allow specifying scripts for additional ports. Events are routed into and out of scripts using the [Select] statement inside a [Pipeline].
```

File: docs/language/scripts.md [hunk 5]
```diff
@@ -4,18 +4,27 @@ sidebar_position: 1
...
+
 Example:
 
 ```tremor
 define script add
 args
     summand
+state
+    # initialize the summand from args
+    state = arts.summand
```
[the remark this answers is not in the document]
Comment (Licenser) [reply docs/language/scripts.md:21]: you are right!!! Shame on me :(

**Review by mfelsche — APPROVED** [review]

**Pull request merged**